meson.build

```diff
@@ -0,0 +1,237 @@
+project('libass', 'c', license: 'ISC',
+        meson_version: '>= 0.55.0',
+        # cpp_std specified because meson compiler options are currently global
+        default_options: ['c_std=c99', 'cpp_std=c++11', 'buildtype=debugoptimized', 'warning_level=2'],
+        version: '0.16.0')
+
+conf = configuration_data()
+deps = []
+
+# Compiler setup
+
+cc = meson.get_compiler('c')
+
+if cc.get_argument_syntax() == 'gcc'
+    cflags = [
+        '-Wno-sign-compare',
+        '-Wno-unused-parameter',
+        '-Werror-implicit-function-declaration',
+        '-Wstrict-prototypes',
+        '-Wpointer-arith',
+        '-Wredundant-decls',
+        '-Wno-missing-field-initializers'
+    ]
+    add_project_arguments(cc.get_supported_arguments(cflags), language: 'c')
+endif
+
+test_args = []
+
+if host_machine.system() == 'linux'
+    test_args += '-D_GNU_SOURCE'
+    add_project_arguments('-D_GNU_SOURCE', language: 'c')
+elif host_machine.system() == 'darwin'
+    test_args += '-D_DARWIN_C_SOURCE'
+    add_project_arguments('-D_DARWIN_C_SOURCE', language: 'c')
+elif cc.get_id() != 'msvc'
+    test_args += '-D_POSIX_C_SOURCE=200809L'
+    add_project_arguments('-D_POSIX_C_SOURCE=200809L', language: 'c')
+endif
+
+# Configuration
+
+if get_option('large-tiles')
+    conf.set('CONFIG_LARGE_TILES', 1)
+endif
+
+conf.set('PACKAGE_NAME', 'libass')
+conf.set('PACKAGE_VERSION', meson.project_version())
+
+conf.set('CONFIG_SOURCEVERSION', '"meson, on or after: ' + meson.project_version() + '"')
+
+str_check_functions = [
+    'strdup',
+    'strndup'
+]
+
+foreach name: str_check_functions
+    if cc.has_function(name) and cc.has_header_symbol('string.h', name, args: test_args)
+        conf.set('HAVE_@0@'.format(name.to_upper()), 1)
+    endif
+endforeach
+
+# Dependencies
+
+deps += cc.find_library('m', required: false)
+
+# Replace with Meson iconv lookup (`dependency('iconv')`) once 0.60 can be required, see: https://github.com/libass/libass/pull/330#discussion_r707666730
+iconv_dep = cc.find_library('iconv', required: false)
+if iconv_dep.found()
+    deps += iconv_dep
+endif
+if iconv_dep.found() or cc.has_function('iconv_open')
+    conf.set('CONFIG_ICONV', 1)
+endif
+
+freetype_dep = dependency('freetype2',version: '>= 2.10.92',
+                          fallback: ['freetype2', 'freetype_dep'],
+                          default_options: ['default_library=' + get_option('default_library'), 'harfbuzz=disabled']) # workaround for https://github.com/mesonbuild/meson/issues/8047
```

Comment:
Need to check if this still allows projects including this to fully resolve the circular dependency via intermediate FreeType. If not it would imho be better to remove `harfbuzz=disabled`.

Also why is this the only one with `'default_library=' + get_option('default_library')`?

Reply:
Meson can't resolve the circular dependency itself; the disabled flag is to ensure this will build despite that. It doesn't apply when fetching a third party build, so won't affect anyone not using the wraps.

The `default_library` flag is related to the linked Meson issue. I don't remember the specifics beyond that at this point.

Reply:
IMHO static vs shared is mostly a user choice, I don't think libass is so special that it has to be the only project doing this. I agree Meson could improve this, but I think it's a bad idea to work around like this. It's just going to make very inconsistent behaviour across projects.

Reply:
@torque can you check if forcing freetype2’s `default_library` is still required and if not remove it?